src/solvers/dg/3d/dg.jl

```diff
@@ -2276,121 +2125,118 @@ function calc_mortar_flux!(surface_flux_values, dg::Dg3D, mortar_type::Val{:l2},
-        surface_flux_values[:, :, :, 5, lower_right_element_id] .= (fstar_lower_right +
-                                                                    noncons_diamond_lower_right)
-      end
-    else # large_sides[m] == 2 -> small elements on left side
-      if dg.l2mortars.orientations[m] == 1
-        # L2 mortars in x-direction
-        surface_flux_values[:, :, :, 2, upper_left_element_id]  .= (fstar_upper_left  +
-                                                                    noncons_diamond_upper_left)
-        surface_flux_values[:, :, :, 2, upper_right_element_id] .= (fstar_upper_right +
-                                                                    noncons_diamond_upper_right)
-        surface_flux_values[:, :, :, 2, lower_left_element_id]  .= (fstar_lower_left  +
-                                                                    noncons_diamond_lower_left)
-        surface_flux_values[:, :, :, 2, lower_right_element_id] .= (fstar_lower_right +
-                                                                    noncons_diamond_lower_right)
-      elseif dg.l2mortars.orientations[m] == 2
-        # L2 mortars in y-direction
-        surface_flux_values[:, :, :, 4, upper_left_element_id]  .= (fstar_upper_left  +
-                                                                    noncons_diamond_upper_left)
-        surface_flux_values[:, :, :, 4, upper_right_element_id] .= (fstar_upper_right +
-                                                                    noncons_diamond_upper_right)
-        surface_flux_values[:, :, :, 4, lower_left_element_id]  .= (fstar_lower_left  +
-                                                                    noncons_diamond_lower_left)
-        surface_flux_values[:, :, :, 4, lower_right_element_id] .= (fstar_lower_right +
-                                                                    noncons_diamond_lower_right)
-      else
-        # L2 mortars in z-direction
-        surface_flux_values[:, :, :, 6, upper_left_element_id]  .= (fstar_upper_left  +
-                                                                    noncons_diamond_upper_left)
-        surface_flux_values[:, :, :, 6, upper_right_element_id] .= (fstar_upper_right +
-                                                                    noncons_diamond_upper_right)
-        surface_flux_values[:, :, :, 6, lower_left_element_id]  .= (fstar_lower_left  +
-                                                                    noncons_diamond_lower_left)
-        surface_flux_values[:, :, :, 6, lower_right_element_id] .= (fstar_lower_right +
-                                                                    noncons_diamond_lower_right)
-      end
-    end
+    @. fstar_upper_left  += noncons_diamond_upper_left
+    @. fstar_upper_right += noncons_diamond_upper_right
+    @. fstar_lower_left  += noncons_diamond_lower_left
+    @. fstar_lower_right += noncons_diamond_lower_right
```

Comment:
Usually we found that `@.` is slower and causes more allocations than hand-written loops. Have you tested the impact of this here (and elsewhere, I saw it a couple of times in the new code)?

Reply:
Micro-benchmarks like
```julia
julia> @benchmark Trixi.calc_mortar_flux!(
           $(dg.elements.surface_flux_values),
           $(dg),
           $(dg.mortar_type),
           $(Trixi.have_nonconservative_terms(dg.equations)),
           $(dg.l2mortars),
           $(dg.thread_cache))
```
 show that both versions use the same allocations. 
```julia
    @. fstar_upper_left  += noncons_diamond_upper_left
    @. fstar_upper_right += noncons_diamond_upper_right
    @. fstar_lower_left  += noncons_diamond_lower_left
    @. fstar_lower_right += noncons_diamond_lower_right
```
takes ca. `42.587μs ` while
```julia
    for j in 1:nnodes(dg), i in 1:nnodes(dg), v in 1:nvariables(dg)
      fstar_upper_left[v, i, j]  += noncons_diamond_upper_left[v, i, j]
    end
    for j in 1:nnodes(dg), i in 1:nnodes(dg), v in 1:nvariables(dg)
      fstar_upper_right[v, i, j] += noncons_diamond_upper_right[v, i, j]
    end
    for j in 1:nnodes(dg), i in 1:nnodes(dg), v in 1:nvariables(dg)
      fstar_lower_left[v, i, j]  += noncons_diamond_lower_left[v, i, j]
    end
    for j in 1:nnodes(dg), i in 1:nnodes(dg), v in 1:nvariables(dg)
      fstar_lower_right[v, i, j] += noncons_diamond_lower_right[v, i, j]
    end
```
takes ca. `41.453 μs `. Is it worth the additional visual clutter? If we are greedy, yes. I'll change that.

Reply:
We *are* greedy, but not just for performance but also for legibility. If possible, I'd use the dot syntax everywhere possible, since it makes the code much easier to read. If in this case the allocations are the same and performance is that similar, just leave it as it is.

TBH, I'd favor using `@.` in even more places if performance permits it...